ipa-core/src/test_fixture/ipa.rs

```diff
@@ -298,22 +298,97 @@ pub async fn test_oprf_ipa<F>(
                     })
                     .collect::<Vec<_>>();
 
-                attribution_and_capping_and_aggregation::<
-                    _,
-                    BA8,  // BreakdownKey,
-                    BA3,  // TriggerValue,
-                    BA20, // Timestamp,
-                    BA5,  // Saturating Sum
-                    Replicated<F>,
-                    F,
-                >(
-                    ctx,
-                    sharded_input,
-                    config.attribution_window_seconds,
-                    ref_to_histogram,
-                )
-                .await
-                .unwrap()
+                if config.per_user_credit_cap == 8 {
```

Comment:
This is horrible. I apologize. I would *love* a suggestion for how to do this in a less terrible way.

@martinthomson, @akoshelev

Reply:
I'm not aware of any.  Basically, what you are really saying here is that based on the value of this variable (`per_user_credit_cap`), run different code.  That the difference between the arms is in many ways a variable, you are now into the realm of metaprogramming.  A macro is one option, but even that carries risks.

The only thing that might make this a miniscule amount easier is to use `match` rather than this if/else business.